src/adapter/http/tests/tests/test_protocol_dataset_helpers.rs

```diff
@@ -120,6 +122,10 @@ async fn test_object_url_local_fs() {
             physical_hash == test_case.data_slice_object().physical_hash &&
             download_from_url == test_case.dataset_url.join(format!("data/{}", physical_hash.as_multibase()).as_str()).unwrap() &&
             download_from_headers == vec![
+                messages::HeaderRow {
+                    name: VersionHeader::name().to_string(),
```

Comment:
Let's add a test for version compatibility check

Reply:
Also would like to see a test that simulates an internal error on the server and checks that client gets the message

Reply:
added